executor/src/witgen/global_constraints.rs

```diff
@@ -268,8 +266,67 @@ fn propagate_constraints<T: FieldElement>(
     (known_constraints, remove)
 }
 
+/// Get simple expression roots for a given expression.
+///
+/// This function recursively evaluates an algebraic expression and attempts
+/// to extract roots from it. The roots are returned as a vector of evaluation
+/// values containing references to algebraic expressions and field elements.
+///
+/// # Parameters
+///
+/// - `expr`: A reference to an `Expression` of type `T`.
+///
+/// # Returns
+///
+/// An `Option` containing a vector of `EvalValue<&AlgebraicReference, T>`
+/// if the roots can be successfully extracted, otherwise `None`.
+pub fn find_algebraic_roots<T: FieldElement>(
```

Comment:
About the name of the function - maybe a bit nitpicking: A root is always algebraic, so I think `find_roots` should be ok.

Reply:
Done,